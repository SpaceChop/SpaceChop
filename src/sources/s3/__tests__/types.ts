import S3SourceConfig from '../types';

describe('validate s3 source', () => {
  it('should throw on invalid source', () => {
    const config = {};
    expect(() => S3SourceConfig.check(config)).toThrowErrorMatchingSnapshot();
  });

  it('should throw on invalid params in source', () => {
    const config = {
      access_key_id: '',
      secret_access_key: '',
      region: '',
      bucket_name: '',
      path: '',
    };
    expect(() => S3SourceConfig.check(config)).toThrowErrorMatchingSnapshot();
  });

  it('should throw on empty path in source', () => {
    const config = {
      paths: ['/:preset/:image'],
      sources: [{ s3: {
        access_key_id: 'access_key_id',
        secret_access_key: 'secret_access_key',
        region: 'region',
        bucket_name: 'bucket_name',
        path: '',
      },
    }], presets: {} };
    expect(() => S3SourceConfig.check(config)).toThrowErrorMatchingSnapshot();
  });

  it('should not throw on valid source', () => {
    const config = {
      access_key_id: 'access_key_id',
      secret_access_key: 'secret_access_key',
      region: 'region',
      bucket_name: 'bucket_name',
      path: ':image',
    };
    expect(() => S3SourceConfig.check(config)).not.toThrow();
  });
});
