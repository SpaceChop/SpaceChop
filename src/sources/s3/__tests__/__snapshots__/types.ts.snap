// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate s3 source should throw on empty path in source 1`] = `"Expected string, but was undefined"`;

exports[`validate s3 source should throw on invalid params in source 1`] = `""`;

exports[`validate s3 source should throw on invalid source 1`] = `"Expected string, but was undefined"`;
